accounts-db/src/tiered_storage/meta.rs

```diff
@@ -82,28 +112,64 @@ pub trait TieredAccountMeta: Sized {
 }
 
 impl AccountMetaFlags {
-    pub fn new_from(optional_fields: &AccountMetaOptionalFields) -> Self {
+    pub fn new_from(optional_fields: &AccountMetaOptionalFields, lamports: u64) -> Self {
```

Comment:
Agree, I don't like the current approach either since I the meta, flags, and optional fields were created before we had some special packing for lamports.  Let me think more about it.